lib/galaxy/model/__init__.py

```diff
@@ -1967,8 +1991,15 @@ def __init__(self, id, **kwargs):
 class Dataset(StorableObject, RepresentById):
     states = Bunch(NEW='new',
                    UPLOAD='upload',
+                   WAITING='waiting',
+                   LIMITED='limited',
+                   DISPATCHED='dispatched',
+                   SUBMITTED='submitted',
+                   STAGEIN='stagein',
                    QUEUED='queued',
                    RUNNING='running',
+                   STAGEOUT='stageout',
+                   FINISHING='finishing',
```

Comment:
I think all these new states should be added to the `non_ready_states` tuple below.